simuleval/agents/service.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import os
+import json
+import logging
+from tornado import web, ioloop
+from simuleval.data.segments import segment_from_json_string
+from simuleval import options
+
+logger = logging.getLogger("simuleval.agent_server")
+
+
+class SystemHandler(web.RequestHandler):
+    def initialize(self, system):
+        self.system = system
+
+    def get(self):
+        self.write(json.dumps({"info": str(self.system)}))
+
+
+class ResetHandle(SystemHandler):
+    def post(self):
+        self.system.reset()
+
+
+class OutputHandler(SystemHandler):
+    def get(self):
+        output_segment = self.system.pop()
+        self.write(output_segment.json())
+
+
+class InputHandler(SystemHandler):
+    def put(self):
+        segment = segment_from_json_string(self.request.body)
+        self.system.push(segment)
+
+
+def start_agent_service(system):
```

Comment:
Ah this is cool too. So the user just needs to implement the agent (system) and simulEval will create the server and handle the requests. nice!

Reply:
Yes, this is also good for the shared task evaluation.